src/client/shader.cpp

```diff
@@ -588,8 +626,8 @@ ShaderInfo ShaderSource::generateShader(const std::string &name,
 
 	video::IVideoDriver *driver = RenderingEngine::get_video_driver();
 	if (!driver->queryFeature(video::EVDF_ARB_GLSL)) {
-		errorstream << "Shaders are enabled but GLSL is not supported by the driver\n";
-		return shaderinfo;
+		throw ShaderException("Shaders are enabled but GLSL is not supported "
+			"by the driver");
```

Comment:
Now that these error messages are shown directly in the GUI, they should be translated with `gettext`.